packages/opentelemetry-core/test/utils/environment.test.ts

```diff
@@ -176,6 +177,20 @@ describe('environment', () => {
         );
       });
     });
+
+    it('memoized version of getEnv works as expected', () => {
+      mockEnvironment({
+        OTEL_LOG_LEVEL: 'waRn',
+      });
+
+      const getEnvFirstInvocationResult = getEnv().OTEL_LOG_LEVEL;
+      const getEnvSecondInvocationResult = getEnv().OTEL_LOG_LEVEL;
+
+      assert.strictEqual(
+        getEnvFirstInvocationResult,
+        getEnvSecondInvocationResult
+      );
+    });
```

Comment:
This test doesn't really test the memoization. Think you would have to spy the underlying `calculateEnv` to make sure it is only called once.